src/interface/meshblock_data.cpp

```diff
@@ -192,27 +192,28 @@ template <typename T>
 const VariableFluxPack<T> &MeshBlockData<T>::PackVariablesAndFluxesImpl(
     const std::vector<std::string> &var_names, const std::vector<std::string> &flx_names,
     const std::vector<int> &sparse_ids, PackIndexMap *map, vpack_types::UidVecPair *key) {
-  return PackListedVariablesAndFluxes(GetVariablesByName(var_names, sparse_ids),
-                                      GetVariablesByName(flx_names, sparse_ids, true),
-                                      map, key);
+  return PackListedVariablesAndFluxes(
+      GetVariablesByName(var_names, sparse_ids),
+      GetVariablesByName(flx_names, sparse_ids, FluxRequest::OnlyFlux), map, key);
 }
 
 /// Variables and fluxes by Metadata Flags
 template <typename T>
 const VariableFluxPack<T> &MeshBlockData<T>::PackVariablesAndFluxesImpl(
     const Metadata::FlagCollection &flags, const std::vector<int> &sparse_ids,
     PackIndexMap *map, vpack_types::UidVecPair *key) {
-  return PackListedVariablesAndFluxes(GetVariablesByFlag(flags, sparse_ids),
-                                      GetVariablesByFlag(flags, sparse_ids, true), map,
-                                      key);
+  return PackListedVariablesAndFluxes(
+      GetVariablesByFlag(flags, sparse_ids),
```

Comment:
Should this be `GetVariablesByFlag(flags, sparse_ids, FluxRequest::NoFlux)` to maintain backward compatibility? For instance, if you specified `Metadata::OneCopy` as the flags wouldn't this slurp up fluxes as variables when it didn't before?

Reply:
It'll be `NoFlux` by default. But if we changed default arguments down the line, I can see that being an issue. I'll change it.

Reply:
Ah, I see. Sounds good

Reply:
:+1: Done